File: Company/Source/SISE3GWeb/Company.Application.Launcher/Areas/Reports/Scripts/ScoreCredit/ScoreCredit.js
var DataExport;

//$(document).ready(function () {
//    getInitialState();
//});

class ScoreCreditQueries {
    static GetDocumentType(typeDocument) {
        return $.ajax({
            type: "POST",
            url: rootPath + "Reports/ScoreCredit/GetDocumentType",
            data: JSON.stringify({
                typeDocument: typeDocument
            }),
            dataType: "json",
            contentType: "application/json; charset=utf-8"
        });
    }

    static GetHistoryScoreCredit(documenttype, documentNumber) {
        return $.ajax({
            type: "POST",
            url: rootPath + "Reports/ScoreCredit/GetHistoryScoreCredit",
            data: JSON.stringify({
                documentType: documenttype,
                documentNum: documentNumber
            }),
            dataType: "json",
            contentType: "application/json; charset=utf-8"
        });
    }

    static loadPrefix(prefixes) {
        $("#listViewScoreCreditValid").UifListView({ source: null, displayTemplate: "#BusinessTemplate", edit: false, delete: false, customEdit: false, height: 300 });
        //prefixes.forEach(item => {
        //item.Status = ParametrizationStatus.Create;
        prefixes.dateRequest = FormatDate(prefixes.dateRequest)
        $("#listViewScoreCreditValid").UifListView("addItem", prefixes);
        //});
    }
}

class ReportScoreCredit extends Uif2.Page {
    getInitialState() {

        $("#listViewScoreCreditValid").UifListView({
            displayTemplate: "#BusinessTemplate",
            edit: false,
            delete: false,
            customEdit: false,
            height: 300
        });

        ScoreCreditQueries.GetDocumentType("3").done(function (data) {
            if (data.success) {
                $("#ddlDocumentType").UifSelect({ sourceData: data.result });
            }
        });
    }

    bindEvents() {
        $('#inputProcess').on('buttonClick', this.searchProc);
        $('#btnExport').on('click', this.fnExcelReport);
        $('.OnlyNumber').ValidatorKey(ValidatorType.Number);
    }

    searchProc(event, selectedItem) {
        if (!selectedItem || !selectedItem.trim() || selectedItem.length < 3) {
            $.UifNotify('show', { 'type': 'danger', 'message': Resources.Language.ErrorInputSearchCoverage, 'autoclose': true })
            return;
        }
        if ($("#ddlDocumentType").UifSelect("getSelected") == "") {
            $.UifNotify('show', { 'type': 'danger', 'message': Resources.Language.MessageDocumentType, 'autoclose': true })
            return;
        }

        if ($("#listViewScoreCreditValid").UifListView('getData').length > 0) {
            $("#listViewScoreCreditValid").UifListView("clear");
        }
        if ($("#tableScoreCredit").UifDataTable('getData').length > 0) {
            $("#tableScoreCredit").UifDataTable("clear");
        }

        inputSearch = selectedItem;
        ScoreCreditQueries.GetHistoryScoreCredit($("#ddlDocumentType").UifSelect("getSelected"), inputSearch).done(function (data) {
            if (data.data.ErrorTypeService == 0) {
                DataExport = data.data.scoreCredits;
                $('#tableScoreCredit').UifDataTable({
                    sourceData: data.data.scoreCredits,
                    order: false
                });
                ScoreCreditQueries.loadPrefix(data.data.scoreCreditValid);
            }
            else {
                $.UifNotify('show', { 'type': 'danger', 'message': data.data.ErrorDescription[0], 'autoclose': true })
            }
        });
    }

    fnExcelReport() {
        var tab_text = "<table border='1px'>";
        var tab_body = "";
        var tab_header = "";
        var textRange; var j = 0;
        var tab = document.getElementById('tableScoreCredit'); // id of table

        tab_header = "<tr>" +
            "<th>Tipo de Documento</th>" +
            "<th>Número Documento</th>" +
            "<th>Puntaje(Score)</th>" +
            "<th>A1</th>" +
            "<th>A2</th>" +
            "<th>A3</th>" +
            "<th>A4</th>" +
            "<th>A5</th>" +
            "<th>A6</th>" +
            "<th>A7</th>" +
            "<th>A8</th>" +
            "<th>A9</th>" +
            "<th>A10</th>" +
            "<th>A11</th>" +
            "<th>A12</th>" +
            "<th>A13</th>" +
            "<th>A14</th>" +
            "<th>A15</th>" +
            "<th>A16</th>" +
            "<th>A17</th>" +
            "<th>A18</th>" +
            "<th>A19</th>" +
            "<th>A20</th>" +
            "<th>A21</th>" +
            "<th>A22</th>" +
            "<th>A23</th>" +
            "<th>A24</th>" +
            "<th>A25</th>" +
            "<th>Código Respuesta</th>" +
            "<th>Respuesta</th>" +
            "<th>Fecha Consulta</th>" +
            "<th>Es valor por defecto ?</th>" +
            "<th>Nombre de Usuario</th>" +
            "</tr>";
        
        $.each(DataExport, function (i, item) {
            tab_body = tab_body + "<tr>" +
                "<td>" + item.idCardTypeCd + "</td>" +
                "<td>" + item.idCardNo + "</td>" +
                "<td>" + item.a1 + "</td>" +
                "<td>" + item.a2 + "</td>" +
                "<td>" + item.a3 + "</td>" +
                "<td>" + item.a4 + "</td>" +
                "<td>" + item.a5 + "</td>" +
                "<td>" + item.a6 + "</td>" +
                "<td>" + item.a7 + "</td>" +
                "<td>" + item.a8 + "</td>" +
                "<td>" + item.a9 + "</td>" +
                "<td>" + item.a10 + "</td>" +
                "<td>" + item.a11 + "</td>" +
                "<td>" + item.a12 + "</td>" +
                "<td>" + item.a13 + "</td>" +
                "<td>" + item.a14 + "</td>" +
                "<td>" + item.a15 + "</td>" +
                "<td>" + item.a16 + "</td>" +
                "<td>" + item.a17 + "</td>" +
                "<td>" + item.a18 + "</td>" +
                "<td>" + item.a19 + "</td>" +
                "<td>" + item.a20 + "</td>" +
                "<td>" + item.a21 + "</td>" +
                "<td>" + item.a22 + "</td>" +
                "<td>" + item.a23 + "</td>" +
                "<td>" + item.a24 + "</td>" +
                "<td>" + item.a25 + "</td>" +
                "<td>" + item.responseCode + "</td>" +
                "<td>" + item.response + "</td>" +
                "<td>" + item.dateRequest + "</td>" +
                "<td>" + item.isDefaultValue + "</td>" +
                "<td>" + item.userName + "</td>" +
                "</tr>"
        });

        tab_text = tab_text + tab_header + tab_body + "</table>";
        tab_text = tab_text.replace(/<A[^>]*>|<\/A>/g, "");//remove if u want links in your table
        tab_text = tab_text.replace(/<img[^>]*>/gi, ""); // remove if u want images in your table
        tab_text = tab_text.replace(/<input[^>]*>|<\/input>/gi, ""); // reomves input params

        var ua = window.navigator.userAgent;
        var msie = ua.indexOf("MSIE ");

        if (msie > 0 || !!navigator.userAgent.match(/Trident.*rv\:11\./))      // If Internet Explorer
        {
            txtArea1.document.open("txt/html", "replace");
            txtArea1.document.write(tab_text);
            txtArea1.document.close();
            txtArea1.focus();
            sa = txtArea1.document.execCommand("SaveAs", true, "Say Thanks to Sumit.xls");
        }
        else                 //other browser not tested on IE 11
            var sa = window.open('data:application/vnd.ms-excel,' + encodeURIComponent(tab_text), );

        return (sa);
    }
}

File: Company/Source/Underwriting/Services/UnderwritingService/Models/ScoreCredit.cs
using System;
using System.Runtime.Serialization;

namespace Sistran.Company.Application.UnderwritingServices.Models
{
    [DataContract]
    public class ScoreCredit
    {
        /// <summary>
        /// Identificador de Score Credit
        /// </summary>
        [DataMember]
        public int ScoreCreditId { get; set; }

        /// <summary>
        /// Identificador de Individual
        /// </summary>
        [DataMember]
        public int IndividualId { get; set; }

        /// <summary>
        /// Identificador de Tipo de Identificación
        /// </summary>
        [DataMember]
        public int IdCardTypeCode { get; set; }

        /// <summary>
        /// Identificación
        /// </summary>
        [DataMember]
        public string IdCardNo { get; set; }

        /// <summary>
        /// Valor del score
        /// </summary>
        [DataMember]
        public string Score { get; set; }

        /// <summary>
        /// Código de Respuesta
        /// </summary>
        [DataMember]
        public int ResponseCode { get; set; }

        /// <summary>
        /// Respuesta
        /// </summary>
        [DataMember]
        public string Response { get; set; }

        /// <summary>
        /// Fecha de petición
        /// </summary>
        [DataMember]
        public DateTime DateRequest { get; set; }

        /// <summary>
        /// Es valor por defecto
        /// </summary>
        [DataMember]
        public bool? IsDefaultValue { get; set; }

        /// <summary>
        /// Identificador del usuario
        /// </summary>
        [DataMember]
        public int UserId { get; set; }

        /// <summary>
        /// Propiedad A1
        /// </summary>
        [DataMember]
        public string A1 { get; set; }

        /// <summary>
        /// Propiedad A2
        /// </summary>
        [DataMember]
        public string A2 { get; set; }

        /// <summary>
        /// Propiedad A3
        /// </summary>
        [DataMember]
        public string A3 { get; set; }

        /// <summary>
        /// Propiedad A4
        /// </summary>
        [DataMember]
        public string A4 { get; set; }

        /// <summary>
        /// Propiedad A5
        /// </summary>
        [DataMember]
        public string A5 { get; set; }

        /// <summary>
        /// Propiedad A6
        /// </summary>
        [DataMember]
        public string A6 { get; set; }

        /// <summary>
        /// Propiedad A7
        /// </summary>
        [DataMember]
        public string A7 { get; set; }

        /// <summary>
        /// Propiedad A8
        /// </summary>
        [DataMember]
        public string A8 { get; set; }

        /// <summary>
        /// Propiedad A9
        /// </summary>
        [DataMember]
        public string A9 { get; set; }

        /// <summary>
        /// Propiedad A10
        /// </summary>
        [DataMember]
        public string A10 { get; set; }

        /// <summary>
        /// Propiedad A11
        /// </summary>
        [DataMember]
        public string A11 { get; set; }

        /// <summary>
        /// Propiedad A12
        /// </summary>
        [DataMember]
        public string A12 { get; set; }

        /// <summary>
        /// Propiedad A13
        /// </summary>
        [DataMember]
        public string A13 { get; set; }

        /// <summary>
        /// Propiedad A14
        /// </summary>
        [DataMember]
        public string A14 { get; set; }

        /// <summary>
        /// Propiedad A15
        /// </summary>
        [DataMember]
        public string A15 { get; set; }

        /// <summary>
        /// Propiedad A16
        /// </summary>
        [DataMember]
        public string A16 { get; set; }

        /// <summary>
        /// Propiedad A17
        /// </summary>
        [DataMember]
        public string A17 { get; set; }

        /// <summary>
        /// Propiedad A18
        /// </summary>
        [DataMember]
        public string A18 { get; set; }

        /// <summary>
        /// Propiedad A19
        /// </summary>
        [DataMember]
        public string A19 { get; set; }

        /// <summary>
        /// Propiedad A20
        /// </summary>
        [DataMember]
        public string A20 { get; set; }

        /// <summary>
        /// Propiedad A21
        /// </summary>
        [DataMember]
        public string A21 { get; set; }

        /// <summary>
        /// Propiedad A22
        /// </summary>
        [DataMember]
        public string A22 { get; set; }

        /// <summary>
        /// Propiedad A23
        /// </summary>
        [DataMember]
        public string A23 { get; set; }

        /// <summary>
        /// Propiedad A24
        /// </summary>
        [DataMember]
        public string A24 { get; set; }

        /// <summary>
        /// Propiedad A25
        /// </summary>
        [DataMember]
        public string A25 { get; set; }

        /// <summary>
        /// Respuesta del servicio
        /// </summary>
        [DataMember]
        public string Request { get; set; }

        /// <summary>
        /// Is Score
        /// </summary>
        [DataMember]
        public bool IsScore { get; set; }
    }
}
